fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -1756,6 +1762,80 @@ private Bundle processEntriesForMethod(Bundle requestBundle, Bundle responseBund
         }
     }
 
+    /**
+     * Processes a request entry with a request method of Patch.
+     *
+     * @param requestEntry
+     *            the request bundle entry
+     * @param responseEntry
+     *            the response bundle entry
+     * @param responseIndexAndEntries
+     *            the hashmap containing bundle entry indexes and their associated response entries
+     * @param entryIndex
+     *            the bundle entry index of the bundle entry being processed
+     * @param localRefMap
+     *            the map of local references to external references
+     * @param requestURL
+     *            the request URL
+     * @param absoluteUri
+     *            the absolute URI
+     * @param requestDescription
+     *            a description of the request
+     * @param initialTime
+     *            the time the bundle entry processing started
+     * @throws Exception
+     */
+    private void processEntryforPatch(Entry requestEntry, Entry responseEntry, Map<Integer, Entry> responseIndexAndEntries, Integer entryIndex,
+        Map<String, String> localRefMap, FHIRUrlParser requestURL, String absoluteUri, String requestDescription, long initialTime)
+        throws FHIROperationException {
+        FHIRRestOperationResponse ior = null;
+        int resourceIdIndex = 1;
+        int resourceTypeIndex = 0;
+        String resourceType = extractValueFromUrl(requestEntry.getRequest().getUrl().getValue(), resourceTypeIndex);
+        String resourceId = extractValueFromUrl(requestEntry.getRequest().getUrl().getValue(), resourceIdIndex);
+        try {
+            if (resourceType != null && resourceId != null) {
+
+                if (requestEntry.getResource().is(Parameters.class)) {
+
+                    Parameters parameters = requestEntry.getResource().as(Parameters.class);
+                    FHIRPatch patch;
+
+                    patch = FHIRPathPatch.from(parameters);
+
+                    ior = doPatch(resourceType, resourceId, patch, null, null, null);
+                    // Process and replace bundle entry.
+                    Bundle.Entry resultEntry =
+                            setBundleResponseFields(responseEntry, ior.getResource(), ior.getOperationOutcome(), ior.getLocationURI(), ior.getStatus().getStatusCode(), requestDescription, initialTime);
+                    responseIndexAndEntries.put(entryIndex, resultEntry);
+                }
+            }
+        } catch (Exception e) {
+          throw new FHIRRestBundledRequestException(e.getMessage());
+
+        }
+
+    }
+
+    /**
+     * 
+     * @param requestUrl URL from the request of the bundle entry
+     * @param index -For extract the specific field from the URL
+     * @return
+     */
+    private String extractValueFromUrl(String requestUrl, int index) {
```

Comment:
This method is not necessary. The reuqestUrl is parsed in processEntriesForMethod().